### PR TITLE
ref: split up the expression crate

### PR DESCRIPTION
**Pull request by bjchambers**
This moves the traits for defining an expression evaluator into `sparrow-interfaces` and the code for executing a sequence of expressions into `sparrow-expr-execution.`.

This allows flattening the `sparrow-expressions` crate, and only using it as a test dependency in a wide variety of places. It is only needed at execution time.

This is preparation for adding `sparrow-aggregations` to implement the stateful aggregations.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
     "rust-analyzer.cargo.features": "all",
     "editor.formatOnSave": true,
+    "rust-analyzer.linkedProjects": [
```
Comment (jordanrfrazier) [inline .vscode/settings.json:4]: I see: 
```
rust-analyzer.linkedProjects (default: [])

    Disable project auto-discovery in favor of explicitly specified set of projects
```
but I don't really know what that means. Can you explain?

File: crates/sparrow-expressions/Cargo.toml [hunk 2]
```diff
@@ -6,7 +6,7 @@ edition.workspace = true
 license.workspace = true
 publish = false
 description = """
-Execution of expressions over record batches.
+Implementation of stateless expressions over batches.
```
Comment (jordanrfrazier) [inline crates/sparrow-expressions/Cargo.toml:9]: I forgot if we added machete to CI, but did you run it here? Figured maybe `sparrow-physical` wouldn't be needed anymore here

**Review by jordanrfrazier — APPROVED** [review]

File: crates/sparrow-expressions/Cargo.toml (hunk 2 above)
Comment (bjchambers) [reply crates/sparrow-expressions/Cargo.toml:9]: We had added it, but due to the default working directory, it only affected Python. I may change the default back to that.

File: crates/sparrow-expr-execution/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+authors.workspace = true
+edition.workspace = true
+license.workspace = true
+publish = false
+description = """
+Execution of expressions over record batches.
+"""
+
+[dependencies]
+arrow-array.workspace = true
```
Comment (bjchambers) [inline crates/sparrow-expr-execution/Cargo.toml:13]: Execution now has very minimal deps, which is nice.

**Pull request merged**